.github/workflows/ci-backend.yml

```diff
@@ -371,7 +371,7 @@ jobs:
         name: Calculate running time
         needs: [django, async-migrations]
         runs-on: ubuntu-latest
-        if: needs.changes.outputs.backend == 'true'
+        if: ${{ needs.changes.outputs.backend == 'true' && github.repository == 'PostHog/posthog' }}
```

Comment:
Ah, but we already had the `github.repository == 'PostHog/posthog` check on the offending step ("Capture running time to PostHog", which can only run on our repo), and that [wasn't helping](https://github.com/PostHog/posthog/actions/runs/10357445092/job/28670502050?pr=24283). I now realize why: `github.repository` on PRs is the repository that the PR is being made _to_, not the repository of the branch. Hence `github.repository == 'PostHog/posthog` **always** is true on PRs to PostHog.

What we must actually use is `github.event.pull_request.head.repo.full_name` – it's a mouthful, but that's the one that reflects the source repo of PRs. Pushed a change to use this one instead of `github.repository` where relevant (PR checks).